android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java

```diff
@@ -235,6 +240,14 @@ public void messageArrived(String topic, MqttMessage message) throws Exception {
                         }else if (message.toString().equals("false")){
                             driveforwards = false;
                         }
+                    } else if (topic.equals("/smartcar/speedometer")) {
+                        TextView speedometer = (TextView)findViewById(R.id.speedometer);
+
+                        double speedMS = Double.parseDouble(message.toString());
+                        double speedKMH = Math.round((speedMS * 10)*10.0)/10.0;
```

Comment:
Correct calculation here?

Reply:
It wasn't, must have been pretty stressed when I did that.🤣 
Fixed now